providers/implementations/kdfs/hkdf.c

```diff
@@ -375,6 +431,49 @@ static int hkdf_common_get_ctx_params(KDF_HKDF *ctx, OSSL_PARAM params[])
             return 0;
     }
 
+    if ((p = OSSL_PARAM_locate(params, OSSL_KDF_PARAM_DIGEST)) != NULL) {
+        const EVP_MD *md = ossl_prov_digest_md(&ctx->digest);
+
+        if (md == NULL)
+            return 0;
+        else if (!OSSL_PARAM_set_utf8_string(p, EVP_MD_get0_name(md)))
+            return 0;
+    }
+
+    /* OSSL_KDF_PARAM_MODE has multiple parameter types, so look for all instances */
+    p = params;
```

Comment:
Maybe do these getters as a follow up PR?
I am not seeing much actual value/purpose in having these, other than code completeness.

Reply:
The value for me is easier testing that resetting a fixed-digest HKDF preserves the digest.
I've split it into a separate PR (https://github.com/openssl/openssl/pull/27365) but then this current PR will depend on the new one (because of my tests).  Once the new PR is merged, I'll rebase this one on top of it.  Unless you just go ahead and ignore the new PR and merge this one.